skiko/build.gradle.kts

```diff
@@ -105,42 +108,44 @@ val skiaDirProviderForCrossTargets: Map<Pair<OS, Arch>, Provider<File>> = crossT
     }
 }
 
-val wasmCrossCompile = tasks.register<CrossCompileTask>("wasmCrossCompile") {
+val compileWasm = tasks.register<CompileSkikoCppTask>("compileWasm") {
     val osArch = OS.Wasm to Arch.Wasm
 
     val unzipper = skiaDirProviderForCrossTargets[osArch]!!
     dependsOn(unzipper)
     val unpackedSkia = unzipper.get()
 
-    compiler.set("emcc")
-    crossCompileTargetArch.set(osArch.second)
+    compiler.set(compilerForTarget(OS.Wasm, Arch.Wasm))
+    buildTargetOS.set(osArch.first)
+    buildTargetArch.set(osArch.second)
     buildVariant.set(buildType)
 
-    sourceFiles =
-        project.fileTree("src/jsMain/cpp") { include("**/*.cc") } +
-        project.fileTree("src/commonMain/cpp") { include("**/*.cc") }
-    if (skiko.includeTestHelpers) {
-        sourceFiles += project.fileTree("src/commonTest/cpp") { include("**/*.cc") }
-    }
-    outDir.set(project.layout.buildDirectory.dir("out/compile/${buildType.id}-${osArch.first.id}-${osArch.second.id}"))
+    val srcDirs = projectDirs("src/jsMain/cpp", "src/commonMain/cpp") +
+        if (skiko.includeTestHelpers) projectDirs("src/commonTest/cpp") else emptyList()
+    sourceRoots.set(srcDirs)
 
     includeHeadersNonRecursive(projectDir.resolve("src/commonMain/cpp"))
     includeHeadersNonRecursive(skiaHeadersDirs(unpackedSkia))
 
     flags.set(listOf(
-        *skiaPreprocessorFlags()
+        *skiaPreprocessorFlags(),
+        "-DSKIKO_WASM"
     ))
 }
 
-fun registerNativeBridgesTask(os: OS, arch: Arch): TaskProvider<CrossCompileTask> {
-    return tasks.register<CrossCompileTask>("${os.id}_${arch.id}_CrossCompile") {
+fun registerNativeBridgesTask(os: OS, arch: Arch): TaskProvider<CompileSkikoCppTask> {
+    return tasks.register<CompileSkikoCppTask>("${os.id}_${arch.id}_CrossCompile") {
```

Comment:
Maybe in the future it would make sense to extract cross-compilation to separate Gradle plugin and apply it instead.